R/radlb.R

```diff
@@ -307,13 +307,21 @@ radlb <- function(ADSL, # nolint
       !is.na(TRTEDTM) ~ as.numeric(as.Date(.data$TRTEDTM)),
       is.na(TRTEDTM) ~ floor(.data$trtsdt_int + (.data$study_duration_secs) / 86400)
     )) %>%
-    dplyr::mutate(ADTM = as.POSIXct(
-      (sample(.data$trtsdt_int:.data$trtedt_int, size = 1) * 86400),
-      origin = "1970-01-01"
+    dplyr::ungroup()
+
+  ADLB <- ADLB %>% # nolint
+    dplyr::group_by(USUBJID) %>%
+    dplyr::arrange(USUBJID, AVISITN) %>%
+    dplyr::mutate(ADTM = rep(
+      sort(as.POSIXct(
+        sample(.data$trtsdt_int[1]:.data$trtedt_int[1], size = nlevels(AVISIT)) * 86400,
+        origin = "1970-01-01"
+      )),
+      each = n() / nlevels(AVISIT)
     )) %>%
+    dplyr::ungroup() %>%
```

Comment:
@Nolan-Steed this is what I did to order the dates in some datasets. You could do something similar (i.e. group by patient, randomly generate events, then sort the events so they're in order for each patient) for `adtte`. Let me know if you have any questions!